httpx/dispatch/connection.py

```diff
@@ -150,13 +141,21 @@ def __init__(
             # HTTP request. Don't try to upgrade to TLS in this case.
             pass
 
+        self.set_open_connection(http_version, socket=socket, on_release=on_release)
+
+    def set_open_connection(
+        self,
+        http_version: str,
+        socket: BaseSocketStream,
+        on_release: typing.Optional[typing.Callable],
+    ) -> None:
         if http_version == "HTTP/2":
-            self.h2_connection = HTTP2Connection(
+            self.open_connection = HTTP2Connection(
```

Comment:
This is the only place where with reference `HTTP2Connection` directly, so we'll be able to lazily-import it here, and fail if `h2` isn't installed. :+1: